pkg/worker/lifecycle.go

```diff
@@ -128,13 +127,26 @@ func (s *Worker) clearContainer(containerId string, request *types.ContainerRequ
 	go func() {
 		// Allow for some time to pass before clearing the container. This way we can handle some last
 		// minute logs or events or if the user wants to inspect the container before it's cleared.
-		time.Sleep(delay)
+		select {
+		case <-time.After(time.Duration(s.config.Worker.TerminationGracePeriod) * time.Second):
+		case <-s.ctx.Done():
+		}
+
+		// If the container is still running, stop it. This happens when a sigterm is detected.
+		container, err := s.runcHandle.State(context.TODO(), containerId)
+		if err == nil && container.Status == "running" {
```

Comment:
Anyway we can get this constant "running" from runc types or something?

Reply:
Yeah we can. Should I update the other place this isn't using a const or type?

https://github.com/beam-cloud/beta9/blob/main/pkg/worker/runc_server.go#L128

Reply:
yeah I think we should

Reply:
Updated.